### DISCUSSION TITLE
How to add sortBy with date dropdown?

### DISCUSSION THREAD

**Discussion by AmitSenjaliya** [algolia/instantsearch#5498] [Q&A]
I want to add a dropdown with two options one is Relevance and the second one is Date (newest).
I have tried many ways but now working after all possible ways.

Here is what I do with algolia instant search.
     

      <div  id="search-input"></div>
      <div id="search-sort-by"></div>
      <div class="" id="search-results"></div>
      <div id="pagination"></div>
     
       const searchClient = algoliasearch(
            "CM8E24234",
            "fe2011b8b05c7c9gdfgdfgdfre"
        );
        const search = instantsearch({
            searchClient,
            indexName: 'Articles'
        });

        search.addWidgets([
            
            instantsearch.widgets.sortBy({
                container: '#search-sort-by',
                items: [
                  {value: 'Articles', label: 'Relevance'},
                  {value: 'postdate_timestamp', label: 'Newest first'}
                ]
              }),

            instantsearch.widgets.searchBox({
                container: '#search-input',
            }),

            instantsearch.widgets.hits({
                container: '#search-results',
                templates: {
                    empty: 'No results',
                    item: '{{ title }}'
                },
            }),

            instantsearch.widgets.pagination({
                container: '#pagination',
            }),
        ]);`

Everything working fine but sortBy not working.
someone suggested creating replicas so I created "postdate_timestamp" replicas.
When I change dropdown the the following error throws in the browser network response:

`Index postdate_timestamp-desc does not exist`

**Comment by Haroenv** [comment]
Did you create that index?

**Reply by AmitSenjaliya** [reply]
@Haroenv 
Yes created the Index name "Articles" and added searchable attributes as "postdate_timestamp" and added ranking-and-sorting as "postdate_timestamp".

**Reply by Haroenv** [reply]
but then the index name is `Articles`, not `postdate_timestamp`. You can have an articles index, and then a separate replica where you change the sorting called `postdate_timestamp`, which can be used in the sort by widget
